webhook/resources/volume/validator.go

```diff
@@ -156,6 +164,53 @@ func (v *volumeValidator) Update(request *admission.Request, oldObj runtime.Obje
 	return nil
 }
 
+func (v *volumeValidator) validateExpansionSize(oldVolume *longhorn.Volume, newVolume *longhorn.Volume) error {
+	oldSize := oldVolume.Spec.Size
+	newSize := newVolume.Spec.Size
+	if newSize == oldSize {
+		return nil
+	}
+	if newSize < oldSize && !newVolume.Status.ExpansionRequired {
+		return fmt.Errorf("shrinking volume %v size from %v to %v is not supported", newVolume.Name, oldSize, newSize)
+	}
+
+	newKubernetesStatus := &newVolume.Status.KubernetesStatus
+	namespace := newKubernetesStatus.Namespace
+	pvcName := newKubernetesStatus.PVCName
+	if pvcName == "" || newKubernetesStatus.LastPVCRefAt != "" {
+		return nil
+	}
+
+	pvc, err := v.ds.GetPersistentVolumeClaim(namespace, pvcName)
+	if err != nil {
+		return err
+	}
+
+	pvcSCName := *pvc.Spec.StorageClassName
+	pvcStorageClass, err := v.ds.GetStorageClassRO(pvcSCName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "cannot find storage class %v", pvcSCName)
+		}
+		return err
+	}
```

Comment:
Just missed this..

This is not right. Just return err, good enough. 
```go
	if err != nil {
		return err
	}
```

Also, `!apierrors.IsNotFound(err)` means `resource found` instead..

Reply:
Removed.
Yes, you are right. Need to be more careful.
Many thanks.